shared/packages/expectationManager/src/internalManager/lib/trackedWorkerAgents.ts

```diff
@@ -46,9 +49,19 @@ export class TrackedWorkerAgents {
 		// workers, but instead just ask until we have got an enough number of available workers.
 
 		let hasQueriedAnyone = false
+		const criticalWorkerPool = (1 - criticalWorkerPoolRatio) * workerAgents.length
+		const criticalWorkerPoolFirstIndex = Math.floor(criticalWorkerPool)
 		await Promise.all(
-			workerAgents.map(async ({ workerId, workerAgent }) => {
+			workerAgents.map(async ({ workerId, workerAgent }, workerIndex) => {
 				if (!workerAgent.connected) return
+				if (!trackedExp.exp.statusReport.requiredForPlayout && workerIndex >= criticalWorkerPoolFirstIndex) {
+					trackedExp.availableWorkers.delete(workerId)
+					trackedExp.noAvailableWorkersReason = {
+						user: 'Worker reserved for requiredForPlayout expectations',
+						tech: `Worker "${workerId}" is within the pool of critical expectation workers, currently that size is ${criticalWorkerPool} out of ${workerAgents.length} total`,
+					}
+					return
+				}
```

Comment:
This logic relies on a false assumption that all workers are equal.
For example, there might be a situation where only the first workers support a (non-requiredForPlayout) expectation (this is is checked in `doYouSupportExpectation` just below), and so that non-requiredForPlayout expectation will never finish.


I think that we should instead of letting the ExpectationManager pick workers, we should instead let the AppContainer spin up a pool of workers and tell them to only handle playout-critical expectations.